src/encord_active/lib/encord/actions.py

```diff
@@ -211,18 +231,56 @@ def create_project(
 
         return new_project
 
+    def fix_pickle_files(self, embedding_type: EmbeddingType, renaming_map):
+        def fix_pickle_file(up, renaming_map):
+            up["label_row"] = renaming_map[up["label_row"]]
+            up["data_unit"] = renaming_map[up["data_unit"]]
+            url_without_extension, extension = up["url"].split(".")
+            changed_parts = [renaming_map[x] if x in renaming_map else x for x in url_without_extension.split("/")]
+            up["url"] = "/".join(changed_parts) + "." + extension
+            return up
+
+        collection = load_collections(embedding_type, self.project_file_structure.embeddings)
+        updated_collection = [fix_pickle_file(up, renaming_map) for up in collection]
+        save_collections(embedding_type, self.project_file_structure.embeddings, updated_collection)
+
+    def replace_uids(self, file_mappings, project_hash):
+        renaming_map = {self.original_project_hash: project_hash}
+
+        for (old_lr, old_du), (new_lr, new_du) in file_mappings.items():
+            os.rename(
+                (self.project_file_structure.data / old_lr / "images" / old_du).as_posix() + ".jpg",
+                (self.project_file_structure.data / old_lr / "images" / new_du).as_posix() + ".jpg",
+            )
+            renaming_map[old_lr], renaming_map[old_du] = new_lr, new_du
+
+        dir_renames = {old_lr: new_lr for (old_lr, old_du), (new_lr, new_du) in file_mappings.items()}
+
+        for (old_lr, new_lr) in dir_renames.items():
+            os.rename(self.project_file_structure.data / old_lr, self.project_file_structure.data / new_lr)
```

Comment:
[nitpick] 💅 

This is slightly more concise IMO

```suggestion
        for (old_lr, old_du), (new_lr, new_du) in file_mappings.items():
            lr_path = self.project_file_structure.data / old_lr
            (lr_path / "images" / f"{old_du}.jpg").rename(lr_path / "images" / f"{new_du}.jpg")
            lr_path.rename(lr_path.parent / new_lr)
            renaming_map[old_lr], renaming_map[old_du] = new_lr, new_du
```

Reply:
haven't tested it but so might not work as is

Reply:
I took some elements from this, thanks!